e2e_test/e2e_transfer_test.go

```diff
@@ -0,0 +1,210 @@
+package e2e
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	ethereum "github.com/celo-org/celo-blockchain"
+	"github.com/celo-org/celo-blockchain/common"
+	"github.com/celo-org/celo-blockchain/common/hexutil"
+	"github.com/celo-org/celo-blockchain/core/types"
+	"github.com/celo-org/celo-blockchain/ethclient"
+	"github.com/celo-org/celo-blockchain/internal/ethapi"
+	"github.com/celo-org/celo-blockchain/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	oneCelo = 1000000000000000000
+)
+
+// TestTransferCELO checks following accounts:
+// - Sender account has transfer value, transaction fee deducted
+// - Receiver account has transfer value added.
+// - Governance account has base fee added.
+// - validator account has tip fee added.
+func TestTransferCELO(t *testing.T) {
+	ac := test.AccountConfig(1, 3)
+	gc, ec, err := test.BuildConfig(ac)
+	gc.Hardforks.EspressoBlock = big.NewInt(0)
+	require.NoError(t, err)
+	network, shutdown, err := test.NewNetwork(ac, gc, ec)
+	require.NoError(t, err)
+	defer shutdown()
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+	defer cancel()
+
+	node := network[0] // validator node
+	client := node.WsClient
+	devAccounts := test.Accounts(ac.DeveloperAccounts(), gc.ChainConfig())
+	sender := devAccounts[0]
+	recipient := devAccounts[1]
+	gateWayFeeRecipient := devAccounts[2]
+
+	// Get datum to set GasPrice/MaxFeePerGas/MaxPriorityFeePerGas to sensible values
+	header, err := network[0].WsClient.HeaderByNumber(ctx, common.Big0)
+	require.NoError(t, err)
+	datum, err := network[0].Eth.APIBackend.GasPriceMinimumForHeader(ctx, nil, header)
+	require.NoError(t, err)
+
+	testTransactionArgs := []*ethapi.TransactionArgs{
```

Comment:
One of the problems with having an array of args like this is that when the test fails you get a line number in the for loop down below, but you have no idea which case failed. And then to pin it down you need to either debug which can be quite slow, or modify the test to add some output or comment out sets of args to find the failing case. So it's much nicer if you can put the for loop contents into a method and then just call that method multiple times within the main test. When it fails you know exactly what failed because you have a stack trace leading you to the call which in turn lets you see immediately which arguments were used.